src/agent.ts

```diff
@@ -0,0 +1,77 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { createFinding } from "./findings";
+import {
+  NETHERMIND_DEPLOYER_ADDRESS as DEPLOYER_ADDRESS,
+  AGENT_REGISTRY_ADDRESS,
+  CREATE_AGENT_ABI,
+  UPDATE_AGENT_ABI,
+  DISABLE_AGENT_ABI,
+} from "./constants";
+
+const provideHandleTransaction = (
+  deployerAddress: string,
+  agentRegistryAddress: string,
+  createAgentAbi: string,
+  updateAgentAbi: string,
+  disableAgentAbi: string
+): HandleTransaction => {
+  return async (txEvent: TransactionEvent) => {
+    const findings: Finding[] = [];
+
+    if (txEvent.from !== deployerAddress.toLowerCase()) return findings;
+
+    // filter by relevant events
+    const agentDeploymentEvents = txEvent.filterFunction(
+      [createAgentAbi, updateAgentAbi, disableAgentAbi],
+      agentRegistryAddress
+    );
+
+    // if no bot deployment events are found
+    if (!agentDeploymentEvents.length) return findings;
+
+    agentDeploymentEvents.forEach((event) => {
```

Comment:
```suggestion
    agentDeploymentCalls.forEach((call) => {
```
As brought up in the other comments.